### ISSUE TITLE
Incorrect Typings for request

### ISSUE THREAD

**Issue by JohnnyMorganz** [auth0/nextjs-auth0#111]
### Description

With the recent updates to the library, the types for all the request parameters in the handlers & getSession & tokenCache were changed from http.IncomingMessage to NextApiRequest eg. https://github.com/auth0/nextjs-auth0/blob/84298659a635c4c98f0fd3039f1c6595f3e12969/src/instance.ts#L34

This is alright for any API routes, however it causes a type collision if you attempt to use getSession and tokenCache inside `getServerSideProps` or `getInitialProps`, as `context.req` in these scenarios are of the type http.IncomingMessage, as can be seen here:
https://github.com/zeit/next.js/blob/695b26ce575e2564f5a4ed1e3b9e2127958f1ea0/packages/next/next-server/lib/utils.ts#L101

https://github.com/zeit/next.js/blob/695b26ce575e2564f5a4ed1e3b9e2127958f1ea0/packages/next/types/index.d.ts#L88

**Comment by haf** [comment]
Fixed by #113

**Comment by Jerry2001Qu** [comment]
@haf  Is there any way to use this fix while waiting for the merge? I can't seem to figure out how to do it.

**Comment by haf** [comment]
For this specific issue, just add `// @ts-ignore` just above the line where you feed the `Request` value in.

**Comment by Jerry2001Qu** [comment]
Sorry, I don't think I understand what you mean. I'm trying to replicate the code under Getting Started.

For this code segment,

```
Profile.getInitialProps = async ({ req, res }) => {
  if (typeof window === 'undefined') {
    const session = await auth0.getSession(req);
    if (!session || !session.user) {
      res.writeHead(302, {
        Location: '/api/login'
      });
      res.end();
      return;
    }
    return { user: session.user };
  }
};
```

Would I simply add // @ts-ignore as such?

```
Profile.getInitialProps = async ({ req, res }) => {
  if (typeof window === 'undefined') {
    // @ts-ignore
    const session = await auth0.getSession(req);
    if (!session || !session.user) {
      res.writeHead(302, {
        Location: '/api/login'
      });
      res.end();
      return;
    }
    return { user: session.user };
  }
};
```

That doesn't seem to work for me, as auth0.getSession(req) returns null.

**Comment by haf** [comment]
Here's a demo that works well https://github.com/auth0/nextjs-auth0/pull/116 — clone my fork and look in the examples/with-apollo-hasura-ssr folder

**Comment by caribou-code** [comment]
Just started using this package and it's pretty cool! Works great, so thanks for the hard work!

Do we know when we can expect the next release with this typing fix?

**Comment by Widcket** [comment]
Hi everyone, this is now fixed in the new [v1.0.0-beta.0](https://github.com/auth0/nextjs-auth0/releases/tag/v1.0.0-beta.0).
